model/riscv_insts_zcmp.sail

```diff
@@ -0,0 +1,643 @@
+  (0xc, 0b00) <-> "{ra, s0-s7}, 80",
+  (0xc, 0b01) <-> "{ra, s0-s7}, 96",
+  (0xc, 0b10) <-> "{ra, s0-s7}, 112",
+  (0xc, 0b11) <-> "{ra, s0-s7}, 128",
+  (0xd, 0b00) <-> "{ra, s0-s8}, 80",
+  (0xd, 0b01) <-> "{ra, s0-s8}, 96",
+  (0xd, 0b10) <-> "{ra, s0-s8}, 112",
+  (0xd, 0b11) <-> "{ra, s0-s8}, 120",
+  (0xe, 0b00) <-> "{ra, s0-s9}, 96",
+  (0xe, 0b01) <-> "{ra, s0-s9}, 112",
+  (0xe, 0b10) <-> "{ra, s0-s9}, 128",
+  (0xe, 0b11) <-> "{ra, s0-s9}, 144",
+  (0xf, 0b00) <-> "{ra, s0-s11}, 112",
+  (0xf, 0b01) <-> "{ra, s0-s11}, 128",
+  (0xf, 0b10) <-> "{ra, s0-s11}, 144",
+  (0xf, 0b11) <-> "{ra, s0-s11}, 160",
+}
+
+union clause ast = CM_PUSH : (bits(4), bits(2))
+
+mapping clause encdec_compressed = CM_PUSH(rlist, sp54)                if extensionEnabled(Ext_Zcmp)
+  <-> 0b101 @ 0b11000 @ rlist : bits(4) @ sp54 : bits(2) @ 0b10        if extensionEnabled(Ext_Zcmp)
+
+function clause execute (CM_PUSH(rlist, sp54)) = {
+  var bytes : bits(4) = zeros();
+
+  if (sizeof(xlen) == 32) then {
+    bytes = 0x4
+  };
+
+  if (sizeof(xlen) == 64) then {
+    bytes = 0x8
+  };
+
+  reg_list : vector (13,dec, bits(5)) = [0b00000,0b00000,0b00000,0b00000,0b00000,0b00000,0b00000,0b00000,0b00000,0b00000,0b00000,0b00000,0b00000];
+
+  match rlist {
+    0x4  => reg_list = [0b00000, 0b00000, 0b00000, 0b00000, 0b00000, 0b00000, 0b00000, 0b00000, 0b00000, 0b00000, 0b00000, 0b00000, 0b00001],
+    0x5  => reg_list = [0b00000, 0b00000, 0b00000, 0b00000, 0b00000, 0b00000, 0b00000, 0b00000, 0b00000, 0b00000, 0b00000, 0b01000, 0b00001],
+    0x6  => reg_list = [0b00000, 0b00000, 0b00000, 0b00000, 0b00000, 0b00000, 0b00000, 0b00000, 0b00000, 0b00000, 0b01001, 0b01000, 0b00001],
+    0x7  => reg_list = [0b00000, 0b00000, 0b00000, 0b00000, 0b00000, 0b00000, 0b00000, 0b00000, 0b00000, 0b10010, 0b01001, 0b01000, 0b00001],
+    0x8  => reg_list = [0b00000, 0b00000, 0b00000, 0b00000, 0b00000, 0b00000, 0b00000, 0b00000, 0b10011, 0b10010, 0b01001, 0b01000, 0b00001],
+    0x9  => reg_list = [0b00000, 0b00000, 0b00000, 0b00000, 0b00000, 0b00000, 0b00000, 0b10100, 0b10011, 0b10010, 0b01001, 0b01000, 0b00001],
+    0xa  => reg_list = [0b00000, 0b00000, 0b00000, 0b00000, 0b00000, 0b00000, 0b10101, 0b10100, 0b10011, 0b10010, 0b01001, 0b01000, 0b00001],
+    0xb  => reg_list = [0b00000, 0b00000, 0b00000, 0b00000, 0b00000, 0b10110, 0b10101, 0b10100, 0b10011, 0b10010, 0b01001, 0b01000, 0b00001],
+    0xc  => reg_list = [0b00000, 0b00000, 0b00000, 0b00000, 0b10111, 0b10110, 0b10101, 0b10100, 0b10011, 0b10010, 0b01001, 0b01000, 0b00001],
+    0xd  => reg_list = [0b00000, 0b00000, 0b00000, 0b11000, 0b10111, 0b10110, 0b10101, 0b10100, 0b10011, 0b10010, 0b01001, 0b01000, 0b00001],
+    0xe  => reg_list = [0b00000, 0b00000, 0b11001, 0b11000, 0b10111, 0b10110, 0b10101, 0b10100, 0b10011, 0b10010, 0b01001, 0b01000, 0b00001],
+    0xf  => reg_list = [0b11011, 0b11010, 0b11001, 0b11000, 0b10111, 0b10110, 0b10101, 0b10100, 0b10011, 0b10010, 0b01001, 0b01000, 0b00001],
```

Comment:
Why do we need this? Just have a single (immutable) list (not vector) and enumerate the right range based on the rlist?